File: app/src/main/java/com/arjanvlek/oxygenupdater/views/AboutActivity.java
package com.arjanvlek.oxygenupdater.views;

import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.method.LinkMovementMethod;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import com.arjanvlek.oxygenupdater.BuildConfig;
import com.arjanvlek.oxygenupdater.R;

public class AboutActivity extends AppCompatActivity {

    private static final String GOOGLE_PLAY_BASE_URL = "market://details?id=";
    private static final String GOOGLE_PLAY_BROWSER__BASE_URL = "https://play.google.com/store/apps/details?id=";

    @Override
    public void onCreate(Bundle savedInstanceSate) {
        super.onCreate(savedInstanceSate);
        setContentView(R.layout.activity_about);

        // Set the version number of the app in the version number field.
        String versionNumber = BuildConfig.VERSION_NAME;
        TextView versionNumberView = (TextView) findViewById(R.id.aboutVersionNumberView);
        versionNumberView.setText(String.format(getString(R.string.about_version), versionNumber));

        //Make the links in the background story clickable.
        TextView storyView = (TextView) findViewById(R.id.aboutBackgroundStoryView);
        storyView.setMovementMethod(LinkMovementMethod.getInstance());

        // Set onClick listener to Google Play rate button.
        Button rateAppButton = (Button) findViewById(R.id.aboutRateButton);
        rateAppButton.setOnClickListener(v -> {
            final String appPackageName = getPackageName();
            try {
                startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(GOOGLE_PLAY_BASE_URL + appPackageName)));
            } catch (ActivityNotFoundException e) {
                try {
                    startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(GOOGLE_PLAY_BROWSER__BASE_URL + appPackageName)));
                } catch (ActivityNotFoundException e1) {
                    Toast.makeText(getApplicationContext(), getString(R.string.error_unable_to_rate_app), Toast.LENGTH_LONG).show();
                }
            }
        });

    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            // Respond to the action bar's Up/Home button
            case android.R.id.home:
                finish();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

}

File: app/src/main/java/com/arjanvlek/oxygenupdater/activities/AboutActivity.kt
package com.arjanvlek.oxygenupdater.activities

import android.os.Bundle
import android.text.method.LinkMovementMethod
import android.view.MenuItem
import androidx.core.view.isVisible
import androidx.lifecycle.Observer
import com.arjanvlek.oxygenupdater.ActivityLauncher
import com.arjanvlek.oxygenupdater.BuildConfig
import com.arjanvlek.oxygenupdater.R
import com.arjanvlek.oxygenupdater.utils.Utils
import com.arjanvlek.oxygenupdater.viewmodels.AboutViewModel
import kotlinx.android.synthetic.main.activity_about.*
import org.koin.androidx.viewmodel.ext.android.viewModel

class AboutActivity : SupportActionBarActivity() {

    private val aboutViewModel by viewModel<AboutViewModel>()

    public override fun onCreate(savedInstanceSate: Bundle?) = super.onCreate(savedInstanceSate).also {
        setContentView(R.layout.activity_about)

        val activityLauncher = ActivityLauncher(this)

        // Set the version number of the app as the subtitle
        collapsingToolbarLayout.subtitle = getString(R.string.summary_oxygen, BuildConfig.VERSION_NAME)
        // Make the links in the background story clickable.
        backgroundStoryTextView.movementMethod = LinkMovementMethod.getInstance()

        discordButton.setOnClickListener { activityLauncher.openDiscord(this) }
        githubButton.setOnClickListener { activityLauncher.openGitHub(this) }
        websiteButton.setOnClickListener { activityLauncher.openWebsite(this) }
        emailButton.setOnClickListener { activityLauncher.openEmail(this) }
        rateButton.setOnClickListener { activityLauncher.openPlayStorePage(this) }

        // banner is displayed if app version is outdated
        bannerLayout.setOnClickListener { activityLauncher.openPlayStorePage(this) }

        aboutViewModel.fetchServerStatus(Utils.checkNetworkConnection(application)).observe(this, Observer {
            if (!it.checkIfAppIsUpToDate()) {
                updateBannerText(it.latestAppVersion!!)
            }
        })
    }

    override fun onBackPressed() = finish()

    /**
     * Respond to the action bar's Up/Home button
     */
    override fun onOptionsItemSelected(item: MenuItem) = if (item.itemId == android.R.id.home) {
        finish()
        true
    } else {
        super.onOptionsItemSelected(item)
    }

    private fun updateBannerText(latestAppVersion: String) {
        bannerLayout.isVisible = true
        bannerTextView.text = getString(R.string.new_app_version, latestAppVersion)
    }
}
